app/models/manageiq/providers/vmware/discovery.rb

```diff
@@ -4,70 +4,61 @@ module ManageIQ
   module Providers
     module Vmware
       class Discovery
-        SERVER_CONSOLE_PORTS = [902, 912].freeze
-
-        ESX_PORTS = [902, 903].freeze
-
-        VC_PORTS = [
-          [
-            135, # VC < 5.1 or
-            7444 # VC >= 5.1
-          ],
-          [
-            139,  # VC < 5.1 or
-            2012, # VC >= 5.1
-            2013,
-            2014
-          ]
-        ].freeze
-
         def self.probe(ost)
-          # Check if VMware Server
-          ost.hypervisor << :vmwareserver if ManageIQ::NetworkDiscovery::Port.scan_open(ost, SERVER_CONSOLE_PORTS).length == 2
-
-          # First check if we can access the VMware webservice before even trying the port scans.
-          begin
-            require 'VMwareWebService/MiqVimClientBase'
-            MiqVimClientBase.new(ost.ipaddr, "test", "test")
-          rescue => err
-            $log&.debug("Vmware::Discovery: Failed to connect to VMware webservice: #{err}. ip = #{ost.ipaddr}")
-            return
+          # Get Info from VMware (vSphere) Web Service API
+          access_webservice(ost) do |info|
+            if add_hypervisor(ost, info)
+              add_os(ost, info)
+            end
           end
+        rescue StandardError => err
+          _log&.warn("Vmware::Discovery: Failed to connect to VMware webservice: #{err}. ip = #{ost.ipaddr}")
+          return
+        end
 
-          $log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Connected to VMware webservice. Machine is either ESX or VirtualCenter.")
-
-          # Next check for ESX or VC. Since VC shares some port numbers with ESX, we check VC before ESX
-
-          # TODO: See if there is a way we can check ESX first, and without having to
-          #   also check VC, since it is more likely there will be more ESX servers on
-          #   a network than VC servers.
+        # Obtains info about IP address from vSphere Web Service API
+        # @param ost [OpenStruct]
+        def self.access_webservice(ost)
+          require 'VMwareWebService/MiqVimClientBase'
+          vim = MiqVimClientBase.new(ost.ipaddr, "test", "test")
+          info = vim&.sic&.about
 
-          checked_vc = false
-          found_vc = false
+          _log&.debug("Vmware::Discovery: ip = #{ost.ipaddr}, Connected to VMware webservice.")
```

Comment:
I don't think `_log&.` is needed since as long as you include `VMDB::Logging` _log will always be defined